lib/autoyast.pm

```diff
@@ -338,4 +350,160 @@ sub create_report {
     return "$error_summary\n\n$expressions_summary\n";
 }
 
+=head2 detect_profile_directory
+
+ detect_profile_directory(profile => $profile, path => $path)
+
+ Try to detect profile directory (autoyast_opensuse/, autoyast_sle{12,15}/, autoyast_sles11/)
+ and returns its path.
+ # TODO: autoyast_{caasp,kvm,qam,xen}
+
+ $profile is the autoyast profile 'autoinst.xml'.
+ $path is AutoYaST profile path
+
+=cut
+sub detect_profile_directory {
+    my (%args)  = @_;
+    my $profile = $args{profile};
+    my $path    = $args{path};
+
+    my $dir    = "autoyast_";
+    my $regexp = $dir . '\E[^/]+\/';
+
+    if (!$profile && $path !~ /\Q$regexp/) {
+        my $distri = get_required_var('DISTRI');
+        if (is_sle) {
+            $distri .= "s" if is_sle('<12');    # sles11
+            my $major_version = get_required_var('VERSION');
+            $major_version =~ s/-SP.*//;
+            $distri .= $major_version;
+        }
+        $path = "$dir${distri}/$path";
+        record_info('INFO', "Trying to use path with detected folder: '$path'");
```

Comment:
maybe i am wrong here but the single ' are not needed. if you want you can try out

Reply:
it was there before, a way to highlight it.